src/hdbscan.jl

```diff
@@ -0,0 +1,303 @@
+# HDBSCAN Graph
+# edge[i] is a list of edges adjacent to the i-th vertex
+# the second element of HDBSCANEdge is the mutual reachability distance.
+HDBSCANEdge = Tuple{Int, Float64}
+struct HDBSCANGraph
+    edges::Vector{Vector{HDBSCANEdge}}
+    HDBSCANGraph(nv::Integer) = new([HDBSCANEdge[] for _ in 1 : nv])
```

Comment:
```suggestion
    adj_edges::Vector{Vector{HDBSCANEdge}} # adj_edges[i] is a vector of edges adjacent to the i-th vertex
  
    HDBSCANGraph(nv::Integer) = new([HDBSCANEdge[] for _ in 1 : nv])
```

Please add the newline between the field definitions and the inner ctor in all your structure declarations.
I suggest to rename `graph.edges` into `graph.adj_edges`, because if one just read the field name `edges` it could be understood as a list of all edges in the graph.